### PR TITLE
Jimin/eudb

### PR DESCRIPTION
**Pull request by jiminwen-msft**
Change the trouter endpoints for EUDB users. User story: https://skype.visualstudio.com/SPOOL/_workitems/edit/3392702

If the resource identity belongs to EudbCountries, direct it to the EUDB_TROUTER_SERVICE_URL. Also updated the default trouter and registrar URL to use teams endpoints.

Tests:
Unit tests;
Manual tested with resources in USA and France.

### CODE REVIEW

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/CommunicationSignalingClient.java [hunk 1]
```diff
@@ -221,6 +226,8 @@ private void start(Context context, ISkypetokenProvider skypetokenProvider, Stri
         }
 
         CloudType cloudType = NotificationUtils.getUserCloudTypeFromSkypeToken(skypeUserToken);
+        String resourceLocation = NotificationUtils.decodeResourceLocationFromJwtToken(skypeUserToken);
+        boolean isEUDUCountry = resourceLocation != null && countriesEUDB.contains(resourceLocation);
```
Comment (angiurgiu) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/CommunicationSignalingClient.java:230]: typo: isEudbCountry

File: sdk/communication/azure-communication-chat/src/test/java/com/azure/android/communication/chat/implementation/notification/NotificationUtilsTest.java [hunk 2]
```diff
@@ -59,6 +59,30 @@ public void canParseMessageMetadata() {
         assertEquals(metadataMap.get("tags"), "tag1");
     }
 
+    @Test
+    public void canParseTokenForRegion() {
+        final String usaSkypeToken = "eyJhbGciOiJSUzI1NiIsImtpZCI6IjY5RUU4QzJFNjUwNzI1OUJBMEYxREQ3MDU0Njk0QjdDQTI4QkQxQkYiLCJ4NXQiOiJhZTZNTG1VSEpadWc4ZDF3VkdsTGZLS0wwYjgiLCJ0eXAiOiJKV1QifQ.eyJza3lwZWlkIjoiYWNzOjE4OGQ0Y2VhLTBhOWItNDg0MC04ZmRjLTdhNWM3MWZlOWJkMF8wMDAwMDAxYS1kZDM2LTE0OTktNzdkNC0wMjQ4MjIwMDA1YmUiLCJzY3AiOjE3OTIsImNzaSI6IjE2OTU2NzM3NjYiLCJleHAiOjE2OTU3NjAxNjYsInJnbiI6ImFtZXIiLCJhY3NTY29wZSI6ImNoYXQiLCJyZXNvdXJjZUlkIjoiMTg4ZDRjZWEtMGE5Yi00ODQwLThmZGMtN2E1YzcxZmU5YmQwIiwicmVzb3VyY2VMb2NhdGlvbiI6InVuaXRlZHN0YXRlcyIsImlhdCI6MTY5NTY3Mzc2Nn0.OmjphUp-TEze3R4a1mo0puUdn7bYw-rFWheodWoe5H-uwKQkTQ_9BzreUJ4qefrXinNlgyVz7DdHAp-_fyh5EQyN6CTwpZdvE4qqmFexk7LJC53GGtGPJVRM_JIXN6vKMDFFrR6wWqwZ22Urr0LpvB2xQRQJJmd9uSOj5TfilJOZrgGTGC0LuKTGLC6kEsAtbDod7zzclZWiApnsG7hgClx86C3fa7-WxLrR_gvqD77HUQOZoNu13uAAVkpFpC1Dk78IhuJfu_m_yEJK0D0z80y0yWN-WE0WLjfnaJNCk8aFbfJdEmvRBQP_mBHBKMVoRDwVYkvVGcXdDYnankYgCw";
```
Comment (cheukchuen) [inline sdk/communication/azure-communication-chat/src/test/java/com/azure/android/communication/chat/implementation/notification/NotificationUtilsTest.java:64]: Are we able to base64 encode on the fly? For example:
```
String fakeValue = "{\"iss\":\"ACS\",\"iat\": 1608152725,\"exp\": 1739688725,\"aud\": \"\",\"sub\": \"\",\"resourceLocation\": \"unitedstates\"}";
String usaSkypeToken = (
            new StringBuilder(a)
                .append("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.")
                .append(Base64.getEncoder().encodeToString(fakeValue.getBytes()))
                .append(".EMS0ExXqRuobm34WKJE8mAfZ7KppU5kEHl0OFdyree8")).toString();
```
That way we know at a glance that the token contains the property `resourceLocation` and its value is set to `unitedstates`

**Review by cheukchuen — COMMENTED** [review]
Have we tested the changes using EU and non-EU resources and see whether the requests are routed to the correct Trouter endpoint? We could check the Trouter logs for that.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/CommunicationSignalingClient.java [hunk 3]
```diff
@@ -221,6 +226,8 @@ private void start(Context context, ISkypetokenProvider skypetokenProvider, Stri
         }
 
         CloudType cloudType = NotificationUtils.getUserCloudTypeFromSkypeToken(skypeUserToken);
+        String resourceLocation = NotificationUtils.decodeResourceLocationFromJwtToken(skypeUserToken);
```
[the remark this answers is not in the document]
Comment (jiminwen-msft) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/CommunicationSignalingClient.java:230]: Updated in NotificationUtils to remove duplicate logic.

**Comment by jiminwen-msft** [comment]
> Have we tested the changes using EU and non-EU resources and see whether the requests are routed to the correct Trouter endpoint? We could check the Trouter logs for that.

Canada resource logs: https://portal.microsoftgeneva.com/s/2EF12998. Connection is made on https://trouter2-azsc-usea-4-b.trouter.teams.microsoft.com:3443/useractivity/activity, which is non-EU.

German resource logs: https://portal-eu.microsoftgeneva.com/s/242F912. Connection is made on https://pub-ent-euno-03-f.trouter.teams.microsoft.com:3443/useractivity/activity, which is EU.

**Review by cheukchuen — APPROVED** [review]

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/CommunicationSignalingClient.java [hunk 4]
```diff
@@ -221,6 +225,8 @@ private void start(Context context, ISkypetokenProvider skypetokenProvider, Stri
...
+        boolean isEUDBCountry = resourceLocation != null && countriesEUDB.contains(resourceLocation);
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/CommunicationSignalingClient.java:229]: Will the resource location in the JWT always be lowercase? Maybe we should do something like the following if we're not sure:
```suggestion
        boolean isEUDBCountry = resourceLocation != null && countriesEUDB.contains(resourceLocation.toLowerCase());
```

**Review by vcolin7 — APPROVED** [review]
Everything LGTM. I just left a comment on the casing of the regions that could be a part of the JWT. I'll approve to unblock for the time being, but please take a look at my comment.

**Pull request merged**